pkg/tsdb/azuremonitor/credentials.go

```diff
@@ -42,44 +42,37 @@ func getAuthType(cfg *setting.Cfg, jsonData *simplejson.Json) string {
 	}
 }
 
-func getDefaultAzureCloud(cfg *setting.Cfg) (string, error) {
+func getDefaultAzureCloud(cfg *setting.Cfg) string {
 	// Allow only known cloud names
-	cloudName := cfg.Azure.Cloud
+	cloudName := ""
+	if cfg != nil && cfg.Azure != nil {
+		cloudName = cfg.Azure.Cloud
+	}
 	switch cloudName {
 	case azsettings.AzurePublic:
-		return azsettings.AzurePublic, nil
+		return azsettings.AzurePublic
 	case azsettings.AzureChina:
-		return azsettings.AzureChina, nil
+		return azsettings.AzureChina
 	case azsettings.AzureUSGovernment:
-		return azsettings.AzureUSGovernment, nil
-	case azsettings.AzureGermany:
-		return azsettings.AzureGermany, nil
-	case azsettings.AzureCustomized:
-		return azsettings.AzureCustomized, nil
+		return azsettings.AzureUSGovernment
 	case "":
 		// Not set cloud defaults to public
-		return azsettings.AzurePublic, nil
+		return azsettings.AzurePublic
 	default:
-		err := fmt.Errorf("the cloud '%s' not supported", cloudName)
-		return "", err
+		return cloudName
 	}
 }
 
-func normalizeAzureCloud(cloudName string) (string, error) {
+func normalizeAzureCloud(cloudName string) string {
 	switch cloudName {
 	case azureMonitorPublic:
-		return azsettings.AzurePublic, nil
+		return azsettings.AzurePublic
 	case azureMonitorChina:
-		return azsettings.AzureChina, nil
+		return azsettings.AzureChina
 	case azureMonitorUSGovernment:
-		return azsettings.AzureUSGovernment, nil
-	case azureMonitorGermany:
-		return azsettings.AzureGermany, nil
-	case azureMonitorCustomized:
```

Comment:
Just found that `azsettings.AzureCustomized` is removed from Azure SDK. Actually no decision has been made about what action to take next.

If no name is specialized for customized cloud,

1. Would it be too loose? E.g., a typo of cloud name will fall into the customized cloud unexpectedly.
1. The constant `azureMonitorCustomized` could be removed.

Reply:
you are probably right, better to keep the old `azsettings.AzureCustomized` for giving a better error if the user has a typo.

Let me update the sdk and add back this key.